patches cause an error with $patch:delete in files with multiple patches

### What happened?

_patchesStrategicMerge_ was used for include a code that declare _$path:delete_ in patches for rewrite output from bases and granually including some structure per environment (after initial declaration in bases)

After Kustomize version was upgraded kustomize from 4.5.x to **5.3.0** and upgrading to _patches_ from _patchesStrategicMerge_ it cause an error in a files multiple patches with _$patch:delete_

### What did you expect to happen?

Running the command "kustomize build ./overlay/test" shows resources that have not been disabled for overlays.

### How can we reproduce it (as minimally and precisely as possible)?

Put this into bash file and check output with _patches_ and _patchesStrategicMerge_
```
mkdir -p kustomize-test/base
mkdir -p kustomize-test/overlay/dev
mkdir -p kustomize-test/overlay/test

cat <<EOF> kustomize-test/base/kustomization.yaml
resources:
- workflow.yaml
EOF

cat <<EOF> kustomize-test/base/workflow.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: workflow-suffix1
spec:
  template:
    spec:
      containers:
      - name: worker
        image: worker:1.14.2
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: workflow-suffix2
spec:
  template:
    spec:
      containers:
      - name: worker
        image: worker:1.14.2
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: workflow-suffix3
spec:
  template:
    spec:
      containers:
      - name: worker
        image: worker:1.14.2
EOF

cat <<EOF> kustomize-test/overlay/test/workflow-patch.yaml
\$patch: delete
apiVersion: apps/v1
kind: Deployment
metadata:
  name: workflow-suffix1
---
\$patch: delete
apiVersion: apps/v1
kind: Deployment
metadata:
  name: workflow-suffix2
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: workflow-suffix3
EOF

cat <<EOF> kustomize-test/overlay/test/kustomization.yaml
resources:
  - ../../base

patches: # instead of patchesStrategicMerge
  - path: workflow-patch.yaml

# patchesStrategicMerge:
#   - workflow-patch.yaml
EOF

kustomize build kustomize-test/overlay/test
```

### Expected output

a proper output with _workflow-suffix3_ deployment declaration

### Actual output

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9902cb]

goroutine 1 [running]:
sigs.k8s.io/kustomize/kyaml/yaml.(*RNode).Content(...)
	sigs.k8s.io/kustomize/kyaml/yaml/rnode.go:707
sigs.k8s.io/kustomize/kyaml/yaml.(*RNode).getMapFieldValue(0x14002260b08?, {0x10476bfb1?, 0x7?})
	sigs.k8s.io/kustomize/kyaml/yaml/rnode.go:420 +0x54
sigs.k8s.io/kustomize/kyaml/yaml.(*RNode).GetApiVersion(...)
	sigs.k8s.io/kustomize/kyaml/yaml/rnode.go:402
sigs.k8s.io/kustomize/kyaml/resid.GvkFromNode(0x140017648b8?)
	sigs.k8s.io/kustomize/kyaml/resid/gvk.go:32 +0x40
sigs.k8s.io/kustomize/api/resource.(*Resource).GetGvk(...)
	sigs.k8s.io/kustomize/api/resource/resource.go:57
sigs.k8s.io/kustomize/api/resource.(*Resource).CurId(0x1400044e960)

### Kustomize version

kustomize 5.3.0

### Operating system

Linux

--- a/api/krusty/multiplepatch_test.go
+++ b/api/krusty/multiplepatch_test.go
@@ -1282,6 +1282,72 @@ metadata:
 	}
 }
 
+func TestSinglePatchWithMultiplePatchDeleteDirectives(t *testing.T) {
+	th := kusttest_test.MakeHarness(t)
+	makeCommonFilesForMultiplePatchTests(th)
+	th.WriteF("overlay/staging/deployment-patch1.yaml", `
+$patch: delete
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx
+---
+$patch: delete
+apiVersion: v1
+kind: Service
+metadata:
+  name: nginx
+`)
+	th.WriteF("overlay/staging/deployment-patch2.yaml", `
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: configmap-in-base
+data:
+  foo2: bar2
+`)
+	th.WriteK("overlay/staging", `
+namePrefix: staging-
+commonLabels:
+  env: staging
+patches:
+  - path: deployment-patch1.yaml
+  - path: deployment-patch2.yaml
+resources:
+  - ../../base
+configMapGenerator:
+- name: configmap-in-overlay
+  literals:
+  - hello=world
+`)
+	m := th.Run("overlay/staging", th.MakeDefaultOptions())
+	th.AssertActualEqualsExpected(m, `
+apiVersion: v1
+data:
+  foo: bar
+  foo2: bar2
+kind: ConfigMap
+metadata:
+  annotations:
+    note: This is a test annotation
+  labels:
+    app: mynginx
+    env: staging
+    org: example.com
+    team: foo
+  name: staging-team-foo-configmap-in-base-8cmgkm9f44
+---
+apiVersion: v1
+data:
+  hello: world
+kind: ConfigMap
+metadata:
+  labels:
+    env: staging
+  name: staging-configmap-in-overlay-dc6fm46dhm
+`)
+}
+
 func TestMultiplePatchesBothWithPatchDeleteDirective(t *testing.T) {
 	th := kusttest_test.MakeHarness(t)
 	makeCommonFilesForMultiplePatchTests(th)

--- a/api/resmap/reswrangler.go
+++ b/api/resmap/reswrangler.go
@@ -181,6 +181,10 @@ func (m *resWrangler) GetMatchingResourcesByAnyId(
 	matches IdMatcher) []*resource.Resource {
 	var result []*resource.Resource
 	for _, r := range m.rList {
+		if r.RNode.IsNilOrEmpty() {
+			continue
+		}
+
 		for _, id := range append(r.PrevIds(), r.CurId()) {
 			if matches(id) {
 				result = append(result, r)

